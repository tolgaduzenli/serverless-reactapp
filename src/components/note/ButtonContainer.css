.button-container{
    display:flex;
    flex-direction:row;
    justify-content: center;
    align-items: center;
}

.button-container > button{
    margin: 10px;
}