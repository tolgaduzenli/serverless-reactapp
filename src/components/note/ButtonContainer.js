import React from 'react'
import './ButtonContainer.css'
class ButtonContainer extends React.Component {
    render() {
        const { onClickDelete, onClickEdit } = this.props
        return <div className="button-container">
            <button className="waves-effect waves-light btn-small orange lighten-2" onClick={onClickEdit}>Edit</button>
            <button className="waves-effect waves-light btn-small red lighten-1" onClick={onClickDelete}>Delete</button>
        </div>
    }
}

export default ButtonContainer