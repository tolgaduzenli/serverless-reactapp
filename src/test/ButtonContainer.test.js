import React from 'react'
import enzyme from 'enzyme'
import ButtonContainer from '../components/note/ButtonContainer'

describe('<ButtonContainer />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = enzyme.shallow(
            <ButtonContainer onClickDelete={jest.fn()}
                onClickEdit={jest.fn()}
            />
        );
    });
    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1)
    })
    it('Should render a <button />', () => {
        expect(wrapper.find('button').length).toEqual(2)
    })
})