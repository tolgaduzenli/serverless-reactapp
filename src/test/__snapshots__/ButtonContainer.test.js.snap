// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer
    onClickDelete={[MockFunction]}
    onClickEdit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="waves-effect waves-light btn-small orange lighten-2"
          onClick={[MockFunction]}
        >
          Edit
        </button>,
        <button
          className="waves-effect waves-light btn-small red lighten-1"
          onClick={[MockFunction]}
        >
          Delete
        </button>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit",
          "className": "waves-effect waves-light btn-small orange lighten-2",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Edit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "className": "waves-effect waves-light btn-small red lighten-1",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="waves-effect waves-light btn-small orange lighten-2"
            onClick={[MockFunction]}
          >
            Edit
          </button>,
          <button
            className="waves-effect waves-light btn-small red lighten-1"
            onClick={[MockFunction]}
          >
            Delete
          </button>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit",
            "className": "waves-effect waves-light btn-small orange lighten-2",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Edit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "waves-effect waves-light btn-small red lighten-1",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
